docs/fundamentals/apicompat/assembly-validation.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Assembly validation
+description: Learn how assembly validation can be used to develop consistent and well-formed multi-targeting assemblies.
+ms.date: 11/30/2023
+ms.topic: overview
+---
+
+# Assembly validation
+
+Similar to [package validation](/dotnet/fundamentals/apicompat/package-validation/overview), assembly validation tooling allows you, as a library developer, to validate that your assemblies are consistent and well formed. It provides the following checks:
+
+- Validates that there are no breaking changes across versions.
+- Validates that the assembly has the same set of public APIs for all the different runtime-specific implementations.
+- Catches any applicability holes.
+
+You can run assembly validation either as an [MSBuild task](#enable-msbuild-task) or using the [Microsoft.DotNet.ApiCompat.Tool global tool](global-tool.md). You can run assembly validation in an inner or outer build for maximum performance. <!--how?--> It's also fully incremental, meaning the comparison is only triggered if the inputs or outputs have changed.
+
+Optionally, you can transform the left and right input assembly paths that are emitted into the suppression file to avoid environment specific paths. To transform the paths, use the [ApiCompatLeftAssembliesTransformationPattern](../../core/project-sdk/msbuild-props.md#apicompatleftassembliestransformationpattern) and [ApiCompatRightAssembliesTransformationPattern](../../core/project-sdk/msbuild-props.md#apicompatrightassembliestransformationpattern) MSBuild properties (or the corresponding options for the command-line tool).
```

Comment:
It would be good to mention that the the input string is expected to be a regex replace pattern. See https://github.com/dotnet/runtime/blob/413efe748ac4c5daabf087ed8ed0d59c511e5cf2/eng/resolveContract.targets#L102-L103 for an example (for the msbuild task)

The item's identity is the regex capture group pattern and the `ReplacementString` metadata is the regex replacement string. That said, this is usually a very advanced scenario.

For the CLI app, the input string expected to have the following format: https://github.com/dotnet/sdk/blob/24e9922c9a9c8f9c004ddecb31634a70eeb5a3e0/src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompat.Tool/Program.cs#L381

Reply:
I want to mention that this really is an advanced scenario so it might be better to put this at the end of the document.